lib/ws_servers/api/algos/algo_worker.js

```diff
@@ -0,0 +1,130 @@
+'use strict'
+
+const debug = require('debug')
+const d = debug('bfx:hf:server:algo-worker')
+
+const { AOHost } = require('bfx-hf-algo')
+const {
+  PingPong, Iceberg, TWAP, AccumulateDistribute, MACrossover, OCOCO
+} = require('bfx-hf-algo')
+const algoOrders = [
+  PingPong, Iceberg, TWAP, AccumulateDistribute, MACrossover, OCOCO
+]
+
+const validateAO = require('./util/validate_ao')
+
+const CHANNEL_MSGS = 'hfpriv'
+
+class AlgoWorker {
+  constructor (settings, algoOrders, bcast) {
+    this.host = null
+    this.userID = null
+
+    this.settings = settings
+    this.algoOrders = algoOrders
+
+    this.pub = bcast.redis ? bcast.redis : bcast.ws
```

Comment:
websocket or redis transport, currently supported in bfx-hf-server: websocket

hosted version might use redis